mcstatus/status_response.py

```diff
@@ -0,0 +1,640 @@
+    """Class for storing JavaServer data from a status response.
+
+    :param icon: Icon of the server. Can be unset. BASE64 encoded.
+    """
+
+    class Players(JavaStatusPlayers):
+        """Deprecated class for `players` field.
+
+        Use `JavaStatusPlayers` instead.
+        """
+
+        class Player(JavaStatusPlayer):
+            """Deprecated class for player in `list` field.
+
+            Use `JavaStatusPlayer` instead.
+            """
+
+            @property
+            @deprecated(replacement="uuid", date="2022-08")
+            def id(self):
+                return self.uuid
+
+            @overload
+            def __init__(self, raw: Dict[str, Any]) -> None:
+                ...
+
+            @overload
+            def __init__(self, name: str, uuid: int) -> None:
+                ...
+
+            def __init__(self, *args, **kwargs) -> None:
+                try:
+                    bound = Signature(
+                        parameters=[Parameter("raw", Parameter.POSITIONAL_OR_KEYWORD, annotation=Dict[str, Any])]
+                    ).bind(*args, **kwargs)
+
+                    deprecated(
+                        lambda: None,
+                        replacement="mcstatus.status_response.JavaStatusPlayer",
+                        date="2022-08",
+                        display_name="JavaStatusResponse.Players.Player",
+                    )()
+
+                    # build and copy instance from `built` method
+                    super().__init__(**self.build(bound.arguments["raw"]).__dict__)
+                except TypeError:
+                    super().__init__(*args, **kwargs)
+
+        @overload
+        def __init__(self, raw: Dict[str, Any]) -> None:
+            ...
+
+        @overload
+        def __init__(self, online: int, max: int, list: List[JavaStatusPlayer]) -> None:
+            ...
+
+        def __init__(self, *args, **kwargs) -> None:
+            try:
+                bound = Signature(
+                    parameters=[Parameter("raw", Parameter.POSITIONAL_OR_KEYWORD, annotation=Dict[str, Any])]
+                ).bind(*args, **kwargs)
+
+                deprecated(
+                    lambda: None,
+                    replacement="mcstatus.status_response.JavaStatusPlayers",
+                    date="2022-08",
+                    display_name="JavaStatusResponse.Players",
+                )()
+
+                instance = self.build(bound.arguments["raw"]).__dict__
+                instance["list"] = [
+                    self.Player(
+                        name=player.name,
+                        uuid=player.uuid,
+                    )
+                    for player in bound.arguments["sample"]
+                ]
+
+                super().__init__(**instance)
+            except TypeError:
+                super().__init__(*args, **kwargs)
+
+        @property
+        @deprecated(replacement="list", date="2022-08")
+        def sample(self):
+            return self.list
```

Comment:
The original implementation of `sample` used `None` if the list was empty, we should replicate that here
```suggestion
        @property
        @deprecated(replacement="list", date="2022-08")
        def sample(self):
            if len(self.list) == 0:
                return None
            return self.list
```